otherlibs/runtime_events/runtime_events_consumer.c

```diff
@@ -664,6 +780,294 @@ static int ml_lost_events(int domain_id, void *callback_data, int lost_words) {
   return 1;
 }
 
+
+static value user_events_find_callback_list_for_event_type(value callbacks_root,
+                                                          value event) {
+  CAMLparam2(callbacks_root, event);
+  CAMLlocal2(tmp_callback_array, event_type);
+  tmp_callback_array = Field(callbacks_root, 6); /* ev_user_events */
+  uintnat array_length = caml_array_length(tmp_callback_array);
+
+  uintnat event_index;
+  event_type = Field(event, 2);
+  if (Is_block(event_type)) {
+    event_index = Int_val(Field(Field(event_type, 0), 2));
+  } else {
+    event_index = Int_val(event_type);
+  }
```

Comment:
It might be useful for readers just to comment your expectations for what these fields will contain